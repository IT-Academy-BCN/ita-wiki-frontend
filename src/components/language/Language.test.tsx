import { render } from "@testing-library/react";
import { FC } from "react";

const Language: FC = () => {
  return (
    <select
      title="lang"
      className="bg-white py-2 px-4 text-[#808080] rounded-lg border border-white shadow-sm focus:outline-none focus:ring-2 focus:ring-[#808080] focus:border-transparent"
    >
      <option>ES</option>
      <option>EN</option>
    </select>
  );
};

describe("Language component", () => {
  it("should render the Language component", () => {
    const { getByTitle } = render(<Language />);
    const selectElement = getByTitle("lang");
    expect(selectElement).toBeInTheDocument();
  });

  it("should have the correct class names", () => {
    const { getByTitle } = render(<Language />);
    const selectElement = getByTitle("lang");
    expect(selectElement).toHaveClass(
      "bg-white py-2 px-4 text-[#808080] rounded-lg border border-white shadow-sm focus:outline-none focus:ring-2 focus:ring-[#808080] focus:border-transparent",
    );
  });

  it("should render a select element", () => {
    const { container } = render(<Language />);
    const selectElement = container.querySelector("select");
    expect(selectElement).toBeInTheDocument();
  });

  it("should render exactly two options", () => {
    const { getByTitle } = render(<Language />);
    const options = getByTitle("lang").querySelectorAll("option");
    expect(options).toHaveLength(2);
  });

  it("should render options with text 'ES' and 'EN'", () => {
    const { getByTitle } = render(<Language />);
    const options = Array.from(
      getByTitle("lang").querySelectorAll("option"),
    ).map((option) => option.textContent);
    expect(options).toEqual(["ES", "EN"]);
  });

  it("should match the snapshot", () => {
    const { container } = render(<Language />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
