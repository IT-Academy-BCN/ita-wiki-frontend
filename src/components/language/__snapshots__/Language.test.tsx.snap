// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Language component > should match the snapshot 1`] = `
<select
  class="bg-white py-2 px-4 text-[#808080] rounded-lg border border-white shadow-sm focus:outline-none focus:ring-2 focus:ring-[#808080] focus:border-transparent"
  title="lang"
>
  <option>
    ES
  </option>
  <option>
    EN
  </option>
</select>
`;
